src/convnets/convnext.jl

```diff
@@ -10,8 +10,9 @@ Creates a single block of ConvNeXt.
 - `λ`: Init value for [LayerScale](https://arxiv.org/abs/2103.17239)
 """
 function convnextblock(planes, drop_path_rate = 0., λ = 1f-6)
-  γ = λ > 0 ? Flux.ones32(planes) * λ : nothing
-  LayerScale(x) = isnothing(γ) ? x : x .* γ
+  γ = Flux.ones32(planes) * λ
+  scale = λ > 0 ? identity : Diagonal(γ, Flux.Zeros())
+  LayerScale(x) = x .* scale
```

Comment:
Now `scale` can be passed into the `Chain` below. The forward pass of `Diagonal(γ, Flux.Zeros())` will do `x .* γ`.

Reply:
You're right I forgot about the matched types issue. Sorry, @theabhirath you'll need to revert to the `LayerScale` variation (but still keep the changes to avoid the `isnothing` check).